src/sealock.c

```diff
@@ -145,18 +144,20 @@ static int sealock_phase_results_to_results(sealock_state_t* lock) {
 static int sealock_apply_phase_wise_result_correction(sealock_state_t *lock, time_t time, time_t duration,
                                                       zsf_results_t *previous) {
   time_t phase_start = lock->times[lock->current_row];
-  assert(time >= phase_start);
+  time_t phase_end = phase_start + duration;
   time_t skipped_time = time - phase_start;
+  time_t new_phase_len = duration - skipped_time;
+
+  // Should always be true due to update before calculation.
+  assert(time >= phase_start);
+  assert(new_phase_len > 0);
 
   if (skipped_time == 0) {
     // No correction required.
     return SEALOCK_OK;
   }
 
-  if (lock->current_row + 1 < lock->times_len) {
-    time_t phase_end = phase_start + duration;
-    time_t new_phase_len = phase_end - phase_start - skipped_time;
-
+  if (time <= phase_end) {
```

Comment:
Following up my earlier comment for line 204, perhaps here we need to use 
_if (time + dimr_interval <= phase_end)_
instead of 
_if (time <= phase_end)_

This will make sure that handling the last phase occurs before the last phase ends.